arches_lingo/views/trees.py

```diff
@@ -0,0 +1,363 @@
+        }
+
+    def serialize_concept(self, conceptid: str, *, parents=False, children=True):
+        data = {
+            "id": conceptid,
+            "labels": [
+                self.serialize_concept_label(label) for label in self.labels[conceptid]
+            ],
+        }
+        if children:
+            data["narrower"] = [
+                self.serialize_concept(conceptid)
+                for conceptid in self.narrower_concepts[conceptid]
+            ]
+        if parents:
+            path = self.add_broader_concept_recursive([], conceptid)
+            scheme_id, parent_concept_ids = path[0], path[1:]
+            if len(parent_concept_ids) > 1:
+                self.polyhierarchical_concepts.add(conceptid)
+            schemes = [scheme for scheme in self.schemes if str(scheme.pk) == scheme_id]
+            data["parents"] = [self.serialize_scheme(schemes[0], children=False)] + [
+                self.serialize_concept(parent_id, children=False)
+                for parent_id in parent_concept_ids
+            ]
+
+            self_and_parent_ids = set([conceptid] + parent_concept_ids)
+            data["polyhierarchical"] = bool(
+                self_and_parent_ids.intersection(self.polyhierarchical_concepts)
+            )
+
+        return data
+
+    def add_broader_concept_recursive(self, working_parent_list, conceptid):
+        # TODO: sort on sortorder at higher stacklevel once captured in original data.
+        broader_concepts = sorted(self.broader_concepts[conceptid])
+        try:
+            first_broader_conceptid = broader_concepts[0]
+        except IndexError:
+            # TODO: sort here too.
+            schemes = sorted(self.schemes_by_top_concept[conceptid])
+            working_parent_list.insert(0, schemes[0])
+            return working_parent_list
+        else:
+            working_parent_list.insert(0, first_broader_conceptid)
+            return self.add_broader_concept_recursive(
+                working_parent_list, first_broader_conceptid
+            )
+
+    def serialize_concept_label(self, label_tile: dict):
+        lang_code = self.language_concepts[label_tile[CONCEPT_NAME_LANGUAGE_NODE][0]]
+        localized_string_objs = label_tile[CONCEPT_NAME_CONTENT_NODE].values()
+        try:
+            value = next(iter(localized_string_objs))["value"]
+        except (StopIteration, KeyError):
+            value = "Unknown"
+        return {
+            "valuetype": self.human_label_type(label_tile[CONCEPT_NAME_TYPE_NODE]),
+            "language": lang_code,
+            "value": value,
+        }
+
+    def get(self, request):
+        data = {
+            "schemes": [self.serialize_scheme(scheme) for scheme in self.schemes],
+        }
+        # Todo: filter by nodegroup permissions
+        return JSONResponse(data)
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ValueSearchView(ConceptTreeView):
```

Comment:
This is returning some odds results that don't seem to match with the query string in any way. I've tried munging about with `elf.default_sensitivity()` but to no fantastic result. Any ideas?

Reply:
We ended up union'ing this with an `__icontains` search.